utils/results/common.go

```diff
@@ -0,0 +1,612 @@
+	}
+	return &applicability
+}
+
+func GetApplicableCvesStatus(entitledForJas bool, applicabilityScanResults []*sarif.Run, cves ...services.Cve) jasutils.ApplicabilityStatus {
+	if !entitledForJas || len(applicabilityScanResults) == 0 {
+		return jasutils.NotScanned
+	}
+	if len(cves) == 0 {
+		return jasutils.NotCovered
+	}
+	var applicableStatuses []jasutils.ApplicabilityStatus
+	for _, cve := range cves {
+		for _, applicabilityRun := range applicabilityScanResults {
+			if rule, _ := applicabilityRun.GetRuleById(jasutils.CveToApplicabilityRuleId(cve.Id)); rule != nil {
+				status := getApplicabilityStatusFromRule(rule)
+				if status != "" {
+					applicableStatuses = append(applicableStatuses, status)
+				}
+			}
+		}
+	}
+	return getFinalApplicabilityStatus(applicableStatuses)
+
+}
+
+func GetApplicableCveStatus(entitledForJas bool, applicabilityScanResults []*sarif.Run, cves []formats.CveRow) jasutils.ApplicabilityStatus {
+	if !entitledForJas || len(applicabilityScanResults) == 0 {
+		return jasutils.NotScanned
+	}
+	if len(cves) == 0 {
+		return jasutils.NotCovered
+	}
+	var applicableStatuses []jasutils.ApplicabilityStatus
+	for _, cve := range cves {
+		if cve.Applicability != nil {
+			applicableStatuses = append(applicableStatuses, jasutils.ApplicabilityStatus(cve.Applicability.Status))
+		}
+	}
+	return getFinalApplicabilityStatus(applicableStatuses)
+}
+
+func getApplicabilityStatusFromRule(rule *sarif.ReportingDescriptor) jasutils.ApplicabilityStatus {
+	if rule.Properties[jasutils.ApplicabilitySarifPropertyKey] != nil {
+		status, ok := rule.Properties[jasutils.ApplicabilitySarifPropertyKey].(string)
+		if !ok {
+			log.Debug(fmt.Sprintf("Failed to get applicability status from rule properties for rule_id %s", rule.ID))
+		}
+		switch status {
+		case "not_covered":
+			return jasutils.NotCovered
+		case "undetermined":
+			return jasutils.ApplicabilityUndetermined
+		case "not_applicable":
+			return jasutils.NotApplicable
+		case "applicable":
+			return jasutils.Applicable
+		}
+	}
+	return ""
+}
+
+func GetDependencyId(depName, version string) string {
+	return fmt.Sprintf("%s:%s", depName, version)
+}
+
+func GetScaIssueId(depName, version, issueId string) string {
+	return fmt.Sprintf("%s_%s_%s", issueId, depName, version)
+}
+
+// GetUniqueKey returns a unique string key of format "vulnerableDependency:vulnerableVersion:xrayID:fixVersionExist"
+func GetUniqueKey(vulnerableDependency, vulnerableVersion, xrayID string, fixVersionExist bool) string {
+	return strings.Join([]string{vulnerableDependency, vulnerableVersion, xrayID, strconv.FormatBool(fixVersionExist)}, ":")
+}
+
+// Relevant only when "third-party-contextual-analysis" flag is on,
+// which mean we scan the environment folders as well (node_modules for example...)
+// When a certain package is reported applicable, and the evidence found
+// is inside the source code of the same package, we should disqualify it.
+//
+// For example,
```

Comment:
Sorry, the example isn't clear enough. If necessary please add a link in the comment that explains it better...

Reply:
this is existing code, just moved it to another file